workflow/controller/workflowpod.go

```diff
@@ -37,6 +37,16 @@ var (
 		Name:      volumeVarArgo.Name,
 		MountPath: common.VarRunArgoPath,
 	}
+	volumeTmpDirArgo = apiv1.Volume{
+		Name: "tmp-dir-argo",
+		VolumeSource: apiv1.VolumeSource{
+			EmptyDir: &apiv1.EmptyDirVolumeSource{},
+		},
+	}
+	volumeMountTmpDirArgo = apiv1.VolumeMount{
+		Name:      volumeTmpDirArgo.Name,
+		MountPath: common.TmpDirArgoPath,
+	}
```

Comment:
I think here the root directory is being overwritten by `/tmp`, that is why the tests are failing. Perhaps you should try using `subPaths` 
see https://kubernetes.io/docs/concepts/storage/volumes/#using-subpath

Reply:
thanks